search-app/src/cmr/search/api/providers.clj

```diff
@@ -43,6 +45,47 @@
                common-routes/CORS_ORIGIN_HEADER "*"}
      :body provider-holdings-formatted}))
 
+(defn- pull-metadata
+  "Look for and pull up the metadata field from the body returned by the metadata db
+  from search app we want to promote this as the response."
+  [response]
+  ;; response passed through timing function index [0] time index [1] result
+  (let [time-taken (first response)
+        response-body (second response)
+        hits (count response-body)
+        provider-metadata (-> response-body
+                     (as-> body (map :metadata body))
+                     json/generate-string)
+        new-response {}]
+    (assoc new-response :hits hits :took time-taken :items (json/parse-string provider-metadata true))))
+
+(defn- pull-metadata-single-provider
+  "Look for and remove metadata field from the body returned by the metadata db for a
+  specific provider from search app we want to promote this as the response."
+ [response]
+  ;; response passed through timing function index [0] time index [1] result
+(let [time-taken (first response)
+      response-body (second response)
+      hits 1
+      provider-metadata (-> response-body
+                   :body
+                   (json/parse-string true)
+                   (:metadata)
+                   json/generate-string)
+      new-response {}]
+  (if (not= (:status response-body) 200)
+        (json/parse-string (:body response-body) true)
+        (assoc new-response :hits hits :took time-taken :items [(json/parse-string provider-metadata)]))))
```

Comment:
I hopped something was already made, but I guess not. Looks better now from the user's point of view.

Reply:
There is something kind of made but, it's very coupled with the collection route methods and not easily generalizable